lib/DBIx/Class/InflateColumn/JSONValue.pm

```diff
@@ -0,0 +1,45 @@
+package DBIx::Class::InflateColumn::JSONValue;
+
+=pod
+
+DBIx::Class::InflateColumn::JSONValue - encodes any value (string, number, arrray, ...)
+in the form value => {}, then JSON encoded.
+
+=cut
+
+use strict;
+use warnings;
+
+use base qw/DBIx::Class/;
+use Mojo::JSON qw/encode_json decode_json/;
+use Try::Tiny;
+
+# This is a simplified version of DBIx::Class::InflateColumn::DateTime
+
+sub register_column {
+	my ($self, $column, $info, @rest) = @_;
+
+	$self->next::method($column, $info, @rest);
+	return unless $info->{inflate_value};
+
+	$self->inflate_column(
+		$column => {
+			inflate => sub {
+				my $str = shift;
+				# This is a bit of a hack.  It appears that sometimes the deflate isn't called on the values
+				# of type string and number so they don't need to be decoded.
```

Comment:
Did you think to try the `allow_nonref` option for the `JSON` serializer instead of this ugly hack?

Reply:
I had tried this.  It didn't seem to work, but right now, I don't recall why.  I will try using `allow_nonref` again.

Reply:
I realized the trouble with using JSON and allow_nonref was that of parsing empty strings.  I've switched this back to using these two and working on handling empty strings.  This is definitely the better way to go.

Reply:
Sorry for the confusion. I'll try to be more clear.  I think this was all an issue with the InflateColumn package. I have switched over to using FilterColumn, which looks like the way to go.